testflinger_agent/agent.py

```diff
@@ -32,6 +32,11 @@ def __init__(self, client):
         self.set_state("waiting")
         self.advertised_queues = self.client.config.get("advertised_queues")
         self.advertised_images = self.client.config.get("advertised_images")
+        location = self.client.config.get("location")
+        if location:
+            self.client.post_agent_data({"location": location})
+        if self.advertised_queues:
+            self.client.post_agent_data({"queues": self.advertised_queues})
```

Comment:
To double-check, should we always send location and queues together, even if location is not-populated (for example)?

Reply:
We only send it once, so I don't think it makes much sense to send two separate requests. If it's not populated, then I think it's ok to send empty data for it.

Reply:
just be aware that we are sending the same *type* of data - especially for advertised queues!

Reply:
gotcha, will update shortly